.circleci/config.yml

```diff
@@ -176,13 +176,28 @@ commands:
             .\venv\Scripts\activate
             pip install --upgrade --force-reinstall --no-deps .
           shell: powershell.exe
+
+      - run:
+          name: install gymansium[mujoco]
```

Comment:
Why do we need to explicitly install Gymnasium? Shouldn't this dependency be picked up by `pip install .` above?

Also confused why we need the MuJoCo dependency -- do any of our tests use MuJoCo?

Reply:
The main thing is about the mujoco part. It gets involved in the windows environment from `import seals` in the notebook and mainly [here](https://github.com/HumanCompatibleAI/seals/blob/master/src/seals/__init__.py#L35). If I don't do this, the notebook will [fail](https://app.circleci.com/pipelines/github/HumanCompatibleAI/imitation/3907/workflows/58f56c5c-7035-464e-a9d6-335832f75284/jobs/15842?invite=true#step-113-944164_97).

Reply:
It is directly [mentioned](https://github.com/HumanCompatibleAI/seals/blob/master/setup.py#L147) in Seals as well. Not quite sure why it shows up only in the Windows environment and only in notebook.

Reply:
I did some digging here and the TL;DR is that the notebooks were only getting tested on Windows. The [SQIL notebook](https://github.com/HumanCompatibleAI/imitation/blob/8db6bfb0cfd533f6418e662890499670bd6b010b/docs/tutorials/8a_train_sqil_sac.ipynb#L4) depends on HalfCheetah which is a MuJoCo environment, so it makes sense we'd need thiis dependency. We made a decision a while back to have MuJoCo dependency be optional for the codebase since it's large and can be somewhat tricky to install, so I'm loathe to reverse that especially just special-casing it for the Windows CI run. I'd suggest we instead switch out HalfCheetah for something like `MountainCarContinuous` that does not require extra dependencies -- tracking in https://github.com/HumanCompatibleAI/imitation/issues/798

Note the `gym.register` line [here](https://github.com/HumanCompatibleAI/seals/blob/master/src/seals/__init__.py#L35) should succeed even if MuJoCo is not present -- it will only fail when attempting to instantiate those environments. Indeed,

This doesn't error on `unit-test-linux` since we're skipping all notebooks: https://github.com/HumanCompatibleAI/imitation/blob/master/tests/test_examples.py#L43 This was because they were meant to be run in readthedocs after https://github.com/HumanCompatibleAI/imitation/pull/565 This run is happening but the notebooks failing is just a *warning*:
```
/home/docs/checkouts/readthedocs.org/user_builds/imitation/checkouts/latest/docs/tutorials/8a_train_sqil_sac.ipynb: WARNING: Executing notebook failed: CellExecutionError [mystnb.exec]
/home/docs/checkouts/readthedocs.org/user_builds/imitation/checkouts/latest/docs/tutorials/8a_train_sqil_sac.ipynb: WARNING: Notebook exception traceback saved in: /home/docs/checkouts/readthedocs.org/user_builds/imitation/checkouts/latest/_readthedocs/html/reports/tutorials/8a_train_sqil_sac.err.log [mystnb.exec]
```
I've opened https://github.com/HumanCompatibleAI/imitation/issues/799 to track this

Mac we are not running `test_examples.py` either due to a separate bug, I'll address in #797

Reply:
Thank you for the detailed explanation! That makes a lot sense!